schema/time.go

```diff
@@ -50,14 +51,46 @@ var (
 	}
 )
 
+// TimeLayout is a specific set of time formats we white list
+type TimeLayout string
+
 // Time validates time based values
 type Time struct {
+	TimeLayouts []TimeLayout // TimeLayouts is set of time layouts we want to validate
+	layouts     []string
+}
+
+// validTimeLayout Validates user specified time layouts against
+// formats.
+func validTimeLayout(tf TimeLayout) error {
+	for _, layout := range formats {
+		if layout == string(tf) {
+			return nil
+		}
+	}
+	return fmt.Errorf("unknown time layout %q", tf)
```

Comment:
I'll fix it and resubmit tonight. Thanks

On Wed, Jul 27, 2016, 16:23 Olivier Poitrey <notifications@github.com>
wrote:

> In schema/time.go
> <https://github.com/rs/rest-layer/pull/25#discussion_r72542137>:
>
> >  // Time validates time based values
> >  type Time struct {
> > +	TimeLayouts []TimeLayout // TimeLayouts is set of time layouts we want to validate
> > +	layouts     []string
> > +}
> > +
> > +// validTimeLayout Validates user specified time layouts against
> > +// formats.
> > +func validTimeLayout(tf TimeLayout) error {
> > +	for _, layout := range formats {
> > +		if layout == string(tf) {
> > +			return nil
> > +		}
> > +	}
> > +	return fmt.Errorf("unknown time layout %q", tf)
>
> I'm all for flexibility :)
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/rs/rest-layer/pull/25/files/a4225dbea7c8fba5d5f95a356a99f66b3c93c503#r72542137>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACK5U9ix6-Vn4-nRs1H4ZLtqoRGHfcqIks5qZ-h8gaJpZM4JPL4t>
> .
>